src/control-plane/backend/lambda/listen-stack-status/index.ts

```
@@ -0,0 +1,222 @@
+    if (res.Stacks && res.Stacks.length > 0) {
+      return res.Stacks[0];
+    }
+    return;
+  } catch (error) {
+    logger.error('Failed to describe stack: ', { stackId, region, error });
+    return;
+  }
+}
+
+async function getPipeline(pipelineId: string): Promise<any> {
+  try {
+    const type = `PIPELINE#${pipelineId}#latest`;
+    const input: QueryCommandInput = {
+      TableName: clickStreamTableName,
+      IndexName: prefixTimeGSIName,
+      KeyConditionExpression: '#prefix= :prefix',
+      FilterExpression: 'deleted = :d AND #type = :type',
+      ExpressionAttributeNames: {
+        '#prefix': 'prefix',
+        '#type': 'type',
+      },
+      ExpressionAttributeValues: {
+        ':d': false,
+        ':prefix': 'PIPELINE',
+        ':type': type,
+      },
+    };
+    const records: Record<string, NativeAttributeValue>[] = [];
+    for await (const page of paginateQuery({ client: docClient }, input)) {
+      records.push(...page.Items as Record<string, NativeAttributeValue>[]);
+    }
+    if (records.length > 0) {
+      return records[0];
+    }
+    return;
+  } catch (err) {
+    logger.error('Failed to query pipeline: ', { pipelineId, err });
+    return;
+  }
+}
+
+async function updatePipelineStackStatus(projectId: string, pipelineId:string, stackDetails: PipelineStatusDetail[]): Promise<void> {
+  try {
+    const input: UpdateCommandInput = {
+      TableName: clickStreamTableName,
+      Key: {
+        id: projectId,
+        type: `PIPELINE#${pipelineId}#latest`,
+      },
+      UpdateExpression: 'SET #stackDetails = :stackDetails',
+      ExpressionAttributeNames: {
+        '#stackDetails': 'stackDetails',
+      },
+      ExpressionAttributeValues: {
+        ':stackDetails': stackDetails,
+      },
+    };
+    await docClient.send(new UpdateCommand(input));
+  } catch (err) {
+    logger.error('Failed to update pipeline stack status: ', { projectId, pipelineId, err });
```

This swallows this event. I prefer to throw the exception to retry it later.